torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -640,6 +681,172 @@ def _need_prefetch_post_backward_hook(self) -> bool:
         else:
             return False
 
+    @contextlib.contextmanager
+    def set_state_dict_type(self, state_dict_type: StateDictType) -> Generator:
```

Comment:
seems only used in FSDP code and users will call things like `local_state_dict`, should we prefix the impl with `_`?

Reply:
This should be exported. Uses may not call `local_state_dict` directly. In fact, users cannot call `local_state_dict` directly if the FSDP module is wrapped inside another non-FSDP module, e.g., Pyper models. This allow users to do something like:
with state_dict_type(LOCAL_STATE_DICT):
       pyper_model.state_dict()

Reply:
great point!